src/main/java/seedu/address/logic/parser/recipe/RecipeAddCommandParser.java

```diff
@@ -102,11 +103,13 @@ RecipeAddIngredientCommand parseAddIngredient(String args) throws ParseException
      * @throws ParseException if the user input does not conform the expected format
      */
     public RecipeAddStepCommand parseAddStep(String args) throws ParseException {
+        Index recipeIndex;
+        Optional<Index> stepIndex;
+        Step toAdd;
```

Comment:
I think it's better to initialise it at the bottom like the original version, since there is no point in initialising it so early

Reply:
okay i move up cause i thought it was kind of messy